mirage/network.ml

```diff
@@ -58,16 +60,14 @@ module Make (Stack : Mirage_stack_lwt.V4) (Dns : Dns_resolver_mirage.S) = struct
   let addr_of_host dns host =
     match Ipaddr.V4.of_string host with
     | Ok ip -> Lwt.return @@ Ok ip
-    | Error (`Msg error_msg) ->
-      Dns.gethostbyname dns host >|= function
-      | [] -> error "Unknown host %S : %s" host error_msg
-      | addrs ->
-        let rec loop = function
-          | [] -> error "No IPv4 addresses found for %S : %s" host error_msg
-          | Ipaddr.V4 x :: _ -> Ok x
-          | Ipaddr.V6 _ :: xs -> loop xs
-        in
-        loop addrs
+    | Error (`Msg _) ->
+      match Domain_name.of_string host with
+      | Ok dn -> begin
+        Dns.gethostbyname dns (Domain_name.host_exn dn) >>= function
```

Comment:
It looks like this function was trying hard not to throw exceptions. Should we be handling `host_exn` here?

Reply:
I'd suggest to use `Domain_name.host` (this function here already returns `(_, _) result Lwt.t` - no need to use exception-raising functions)

Reply:
Done